src/Icon/index.tsx

```diff
@@ -0,0 +1,52 @@
+import React, {forwardRef} from 'react';
+import {withTheme} from 'styled-components';
+
+import {getSpacing} from '../utils';
+import {LocalThemeSection} from '../theme/types';
+
+import iconMap from './iconMap';
+
+const sizeMap = {
+	unscaled: 0,
+	default: 1,
+	medium: 1.5,
+	big: 3,
+	huge: 6,
+};
+
+interface CustomIconProps {
+	color: string,
+	dimmed: boolean,
+	flip: string,
+	rotate: number,
+	vAlign: boolean,
+
+	name: string,
+	scale: number,
+	size: string,
+	spacingLeft: any, // seems this can take a string or a number, not good!
+	spacingRight: any,
```

Comment:
It's very useful to define either pixel sizes or fractions as a number (`spacingLeft={2}`, `spacingLeft={0.5}`) and theme tokens or percentages as a string (`spacingLeft="xs"` `spacingLeft="50%"`).

Can't we just say 

```suggestion
	spacingLeft: string | number,
	spacingRight: string | number,
```

The `getSpacing` function does lots of type checking to spit out something useful.